### PR TITLE
TRUS-3724 Non Taxable Individual Beneficiary

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/mapping/registration/IndividualBeneficiaryMapper.scala [hunk 1]
```diff
@@ -50,6 +50,25 @@ class IndividualBeneficiaryMapper @Inject()(addressMapper: AddressMapper) extend
...
   }
 
+  def removeIncompleteBeneficiaries(beneficiaries: List[IndividualBeneficiary], userAnswers: UserAnswers): List[IndividualBeneficiary] = {
+    beneficiaries.filter(indDetailsType =>
+      (userAnswers.is5mldEnabled, userAnswers.isTaxable) match {
+        case (true, true) =>
+            indDetailsType.incomeYesNo.isDefined &&
+            indDetailsType.vulnerableYesNo.isDefined &&
+            indDetailsType.countryOfResidence.isDefined &&
+            indDetailsType.countryOfNationality.isDefined &&
```
Comment (jackheslop96) [inline app/mapping/registration/IndividualBeneficiaryMapper.scala:60]: I'm not sure I agree with this. Country of residence and country of nationality can be None if the user answers no to the "do you know the country of nationality/residence?" question

File: app/mapping/registration/IndividualBeneficiaryMapper.scala [hunk 2]
```diff
@@ -50,6 +50,25 @@ class IndividualBeneficiaryMapper @Inject()(addressMapper: AddressMapper) extend
...
+            indDetailsType.mentalCapacityYesNo.isDefined
+        case (true, false) =>
+            indDetailsType.countryOfResidence.isDefined &&
+            indDetailsType.countryOfNationality.isDefined &&
```
Comment (jackheslop96) [inline app/mapping/registration/IndividualBeneficiaryMapper.scala:64]: Same as above

File: app/mapping/registration/IndividualBeneficiaryMapper.scala (hunk 1 above)
Comment (carrielies) [reply app/mapping/registration/IndividualBeneficiaryMapper.scala:60]: Yeah, your correct

File: app/mapping/registration/IndividualBeneficiaryMapper.scala (hunk 2 above)
Comment (carrielies) [reply app/mapping/registration/IndividualBeneficiaryMapper.scala]: updated now

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**